lib/bundler/ui/shell.rb

```diff
@@ -11,6 +11,7 @@ def initialize(options = {})
         end
         @shell = Thor::Base.shell.new
         @level = ENV['DEBUG'] ? "debug" : "info"
+        @deprecation_messages = Hash.new
```

Comment:
Yeah, I think it probably should be. `Set` is [slower than `Hash`](http://blog.x-aeon.com/2012/02/07/using-data-sets-in-ruby-performance-of-hash-vs-set/), but that really shouldn't matter here.